docs/pages/rmf-audio-capture_L3_Test_Procedure.md

```diff
@@ -0,0 +1,170 @@
+# L3 Test Procedure Document
+## Table of Contents
+
+  - [Overview](#overview)
+    - [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+    - [Definitions](#definitions)
+    - [References](#references)
+  - [Level 3 Test Procedure](#level-3-test-procedure)
+    - [UT-Core functions to be added](#ut-core-functions-to-be-added)
+    - [Raft L3 Testing Requirements](#raft-l3-testing-requirements)
+    - [Input Stream Requirements](#input-stream-requirements)
+
+## Overview
+
+This document describes the level 3 testing suite for the RMF_AudioCapture module.
+
+### Acronyms, Terms and Abbreviations
+
+- `HAL` \- Hardware Abstraction Layer, may include some common components
+- `HAL.h`  \- Abstracted defined API to control the hardware
+- `HAL.c`  \- Implementation wrapper layer created by the `OEM` or `SoC` Vendor.
+- `RDK`  \- Reference Design Kit for All Devices
+- `RDK-V`  \- Reference Design Kit for Video Devices
+- `UT`  \- Unit Test(s)
+- `OEM`  \- Original Equipment Manufacture (Sky is also an OEM)
+- `SoC`  \- System on a Chip
+
+### Definitions
+
+- `Broadcom` \- `SoC` manufacturer <https://www.broadcom.com/>
+- `Amlogic` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Amlogic>
+- `Realtek` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Realtek>
+- `Soc Vendor` \- Definition to encompass multiple vendors
+- `Unit Tests` \- C Function tests that run on the target hardware
+- `Common Testing Framework` \- Off the shelf 3rd Party Testing Framework, or framework that does not require infrastructure to control it. That's not to say it cannot be controlled via infrastructure if required. Examples of which are.
+  - `GTest` \- Google Test Suit <https://google.github.io/googletest>
+  - `CUnit` \- C Testing Suit <http://cunit.sourceforge.net/>
+  - `Unity` \- C Embedded Testing Suit <http://www.throwtheswitch.org/unity>
+  - `ut-core` \- Common Testing Framework <https://github.com/comcast-sky/rdk-components-ut-core>, which wraps a open-source framework that can be expanded to the requirements for future comcast-sky framework.
+
+### References
+
+- `Feedback Loops` \- <https://www.softwaretestingnews.co.uk/4-methods-to-improve-your-feedback-loops-and-supercharge-your-testing-process/>
+- `Doxygen` \- SourceCode documentation tool - <https://www.doxygen.nl/index.html>
+- `Black Box Testing` \- <https://en.wikipedia.org/wiki/Black-box_testing>
+
+## Level 3 Test Procedure
+
+## UT-Core functions to be added
+
+- L3 - RMF_AudioCapture_Open - Depricated not being tested
```

Comment:
w've got to L3 doc here, so  headings like requirements like title goals etc, are required.

## Requirements to support L3 from UT-Core

The following functions require being implemented into the UT-Core module to support control of the interface from a higher level. 

May want to fill this sort with thing out, happy to change it if you think it's overkill. -> 

|Title|Details|
|--|--|
|Function Name|`RMF_AudioCapture_Open`|
|Description| Open default configuration into a global variable and leave open  |
|Test Group| 03 |

**Pre-Conditions :** None

**Dependencies :** None

**User Interaction :** None

Reply:
ok we need to discuss this. We need to be more fluid, what I gave you was an example of what needed to be done, not the literal.

We need to think about what is the requirement for the engineer to test his code. That should be in UT_CORE.

What is the requirement to repeat the steps & set up the prerequisite that, should be in python.

Reply:
updated as discussed